### PR TITLE
refact getPublishTopicList method and merge repeat catch statement

### PR DESCRIPTION
**Pull request by xichaodong**
This commit contain two part refact

1.refact getPublishTopicList loop to HashSet constructor
2.merge repeat catch statement 
#3857

### CODE REVIEW

**Comment by ni-ze** [comment]
please add a new issue and associate it to this pr.  A naked PR is not recommended.

**Comment by xichaodong** [comment]
> please add a new issue and associate it to this pr. A naked PR is not recommended.

#3587

**Comment by HScarb** [comment]
The client should support JDK 1.6. See https://github.com/apache/rocketmq/issues/3684

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java [hunk 1]
```diff
@@ -292,12 +292,7 @@ public void shutdown(final boolean shutdownFactory) {
 
     @Override
     public Set<String> getPublishTopicList() {
-        Set<String> topicList = new HashSet<String>();
-        for (String key : this.topicPublishInfoTable.keySet()) {
-            topicList.add(key);
-        }
-
-        return topicList;
+        return new HashSet<>(this.topicPublishInfoTable.keySet());
```
Comment (HScarb) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:295]: Diamond types are not supported at language level '6'.
It should be like
```java
return new HashSet<String>(this.topicPublishInfoTable.keySet());
```

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java [hunk 2]
```diff
@@ -622,14 +617,7 @@ private SendResult sendDefaultImpl(
...
                         }
-                    } catch (RemotingException e) {
-                        endTimestamp = System.currentTimeMillis();
-                        this.updateFaultItem(mq.getBrokerName(), endTimestamp - beginTimestampPrev, true);
-                        log.warn(String.format("sendKernelImpl exception, resend at once, InvokeID: %s, RT: %sms, Broker: %s", invokeID, endTimestamp - beginTimestampPrev, mq), e);
-                        log.warn(msg.toString());
-                        exception = e;
-                        continue;
-                    } catch (MQClientException e) {
+                    } catch (RemotingException | MQClientException e) {
```
Comment (HScarb) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:620]: Multi-catches are not supported at language level '6'

Comment (xichaodong) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java]: ok,i had restore this  exception statement for jdk1.6

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java (hunk 1 above)
Comment (xichaodong) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java]: ok , i had add generic type to hashmap for jdk 1.6

**Review by HScarb — APPROVED** [review]

**Comment by XiaoyiPeng** [comment]
Please make sure set the target branch to **develop** instead of **master**.

**Pull request closed without merging**